app/CustomFields/HeaderDefinition.php

```diff
@@ -41,10 +41,11 @@ public static function get($post_type) {
           'key' => $key . '_background_video',
           'label' => 'Background video',
           'name' => 'background_video',
-          'type' => 'url',
-          'instructions' => 'The absolute URL to the MP4 video file (http://site.com/video.mp4)',
+          'type' => 'file',
           'required' => 1,
-          'default_value' => '',
+          'library' => 'all',
+          'return_format' => 'url',
+          'mime_types' => 'avi, mov, mpg, mp4',
```

Comment:
@iamdeaneyelid Should we even allow avi, mov, mpg? I think just MP4, as we're not doing any transcoding.